src/DonationForms/resources/registrars/templates/fields/Phone.tsx

```diff
@@ -1,15 +1,86 @@
-import InputMask from 'react-input-mask';
-
 import {PhoneProps} from '@givewp/forms/propTypes';
+import {useEffect} from 'react';
+import InputMask from 'react-input-mask';
 
-export default function Phone({Label, ErrorMessage, placeholder, description, phoneFormat, inputProps}: PhoneProps) {
+/**
+ * @unreleased
+ */
+export default function Phone({
+    Label,
+    ErrorMessage,
+    placeholder,
+    description,
+    phoneFormat,
+    inputProps,
+    intlTelInputSettings,
+}: PhoneProps) {
     const FieldDescription = window.givewp.form.templates.layouts.fieldDescription;
+    const {useFormContext} = window.givewp.form.hooks;
+    const {setValue} = useFormContext();
+
+    const intlTelInputId = inputProps.name + '_intl_tel_input';
+    const isIntlTelInput =
+        intlTelInputSettings.cssUrl && intlTelInputSettings.scriptUrl && intlTelInputSettings.utilsScriptUrl;
+
+    useEffect(() => {
+        if (!isIntlTelInput) {
+            return;
+        }
+
+        const input: HTMLInputElement = document.querySelector('#' + intlTelInputId);
+
+        const css = document.createElement('link');
+        css.href = intlTelInputSettings.cssUrl;
+        css.rel = 'stylesheet';
+        document.body.appendChild(css);
+
+        const script = document.createElement('script');
+        script.src = intlTelInputSettings.scriptUrl;
+        script.async = true;
+        script.onload = () => {
```

Comment:
I'm a little confused on what's going on here within the context of a react component.  Maybe i'm missing something, but It feels like reaching into the dom for an input we already have access to directly is unnecessary.  I would expect the intl input to supply it's own css and external scripts without reaching into the dom.

Reply:
I also was expecting to use their react component, but last week it wasn't being imported from `'intl-tel-input/react'` as demonstrated in their [demo page](https://intl-tel-input.com/examples/react-component.html). 

I tried it again today, as they released a few new versions fixing bugs since last week, now the import is working but the component still is pretty buggy and the autoPlaceholder feature isn't working.

So, this was a solution I found to make it work for now, but we can try to update it again in the near future as this library has a very active release frequency as you can see here: https://github.com/jackocnr/intl-tel-input/releases

Reply:
oh yeah, that looks much easier to maintain! ideally we would use that but I trust you if you say it's buggy - I would suggest creating an issue on their repo with what you are finding is not working as expected.

For my feedback I was really wondering why we couldn't just just refs instead of ID's and render script and style tags instead of using createElement like:

```
<input type={'hidden'} placeholder={placeholder} {...inputProps} /><input id={intlTelInputId} type={'text'} aria-invalid={fieldError ? 'true' : 'false'} />
<script src={intlTelInputSettings.scriptUrl} />
```

Also, it looks like we're using a CDN for this library, any reason to use that over bundling the library? if we bundle the library then we would not even need to include script/style tags as we can access from node_modules

Reply:
Thanks for the clarification and explanations about the `useRef`

About this:

> Also, it looks like we're using a CDN for this library, any reason to use that over bundling the library? if we bundle the library then we would not even need to include script/style tags as we can access from node_modules

Initially, I was using it because the `utilsScript` is pretty big, and serving it through the CDN is faster. However, it's not necessary for the other scripts, so I changed it to import from `node_modules`.

So, it was resolved in 8226c6dc942a22ff20fd024c8e359827901e5b76 and 4a7c5b3945a68d416dcac2f52da5037c4ff9fc8c 

However, the `utilsScript` kept being loaded through the CDN. Even because it is not possible to import it from `node_modules` as explained here: https://github.com/jackocnr/intl-tel-input/issues/980#issuecomment-541060928

_Do you think we should implement one of the solutions discussed in the link above to load it without using the CDN?_

Another important thing to notice is that to make the `utilsScript` properly load in the Form Builder and the Form Builder Preview, was necessary to use a timeout. Without this timeout, problems related to the autoPlaceholder feature were happening.

In the form preview, also was necessary to do this: https://github.com/impress-org/givewp/pull/7340/files#diff-cbb7d030d97960550cc8faf9e98582fbd7772afe2fc33d82c79b7b178102e528R43-R50

About this: 

>  I would suggest creating an issue on their repo with what you are finding is not working as expected.

When I made the changes that I mentioned previously I realized that the problems I was facing using the React component were the same related to timeout and form preview that I mentioned above. For some reason, they did not happen before when loading all scripts through the CDN, but it started after I changed it.

So, I was able to make it work as expected in the Donations Form:

```
import IntlTelInput from 'intl-tel-input/react';
import 'intl-tel-input/build/css/intlTelInput.css';
<IntlTelInput
      initOptions={{
          utilsScript: intlTelInputSettings.utilsScriptUrl,
          initialCountry: intlTelInputSettings.initialCountry,
          showSelectedDialCode: intlTelInputSettings.showSelectedDialCode,
          strictMode: intlTelInputSettings.strictMode,
          i18n: intlTelInputSettings.i18n,
      }}
/>
```


But in Form Builder and Form Builder preview, I wasn't able to delay the `utilsScript` loading, so I kept using the `useRef` approach instead of using the React component.

Reply:
@glaubersilva okay sounds like you experimented a lot with this.  My ideal solution would be to use their react component (for less maintenance) and not have to do anything weird to account for the form builder.  I'd really like to understand _why_ the timeout is needed and figure out if there is something we can tweak without compensating.  I would also be in favor of letting the autocomplete functionality be limited at this time as a trade-off. 

Let me know what you think, we could also do some some pair programming as well 😄

Reply:
@jonwaldstein Gotcha, no problem! I changed it to use the React component.

Resolved in ed12a8edef73e664e3e2842abf7a7cda3502e754 and ba59a6e82fe87617b21009cbecd98e8632786195 and 134d68c775f17c3df4fcdaa6700072d87821233f

About this:

> I'd really like to understand why the timeout is needed...

Even using the React component still was necessary to use a timeout inside a `useEffect` to fix these undesired behaviors in the Form Builder:

1. The autoPlaceholder feature doesn't work.
2. The autoPlaceholder feature doesn't use the proper padding left in the first render.

I left a few comments in the mentioned commits above with these details, but you also can check them in the attached screenshots:

![image](https://github.com/impress-org/givewp/assets/16308864/1d7e9558-9298-465b-8834-6df871265cac)
![image](https://github.com/impress-org/givewp/assets/16308864/5a7c98ea-f076-4649-abcb-d489a20c6295)

About this:

> ...and figure out if there is something we can tweak without compensating.

Seems the problem happens due to a delay in the load of the `utilsScript` that only happens in the Form Builder, but I didn't investigate it.  

In fact, I don't think it's a problem from the form builder that we can tweak, but a problem related to how the library loads the `utilsScript` in certain scenarios. 

In the form builder, we are adding the component to a page already loaded, and the `utilsScript` seems to have difficulty getting loaded in this case. But in the Donations Form, it works as expected.